filestorage.go

```diff
@@ -192,39 +186,41 @@ func (fs *FileStorage) Lock(key string) error {
 	fs.fileStorageNameLocks[key] = fw
 	fileStorageNameLocksMu.Unlock()
 
-	var checkedStaleLock bool // sentinel value to avoid infinite goto-ing
+	for {
+		// parent dir must exist
+		if err := os.MkdirAll(lockDir, 0700); err != nil {
+			fileStorageNameLocksMu.Unlock()
```

Comment:
I believe this Unlock will panic, since it was unlocked on line 187/193 above.

Reply:
oops.. fixed